tools/cfn2ts/lib/codegen.ts

```diff
@@ -223,6 +206,36 @@ export default class CodeGenerator {
       this.code.line(`public static readonly REQUIRED_TRANSFORM = ${JSON.stringify(spec.RequiredTransform)};`);
     }
 
+    //
+    // The static cfnProps2CdkProps() method,
+    // used in the @aws-cdk/cfn-include module
+    //
+
+    this.code.line();
+    this.code.line('/**');
+    this.code.line(' * A method that translates between CloudFormation properties');
+    this.code.line(' * (which use UpperCamelCase property names)');
+    this.code.line(' * and CDK properties');
+    this.code.line(' * (which use lowerCamelCase property names).');
+    this.code.line(' * Used by the @aws-cdk/cfn-include module.');
+    this.code.line(' *');
+    this.code.line(' * @experimental');
+    this.code.line(' */');
+    // if propsType is undefined, we don't use the argument,
+    // and so it has to be prefixed with an underscore,
+    // otherwise the TypeScript compiler settings generated by JSII
+    // will fail compilation
+    const propertiesArg = propsType ? 'properties' : '_properties';
+    this.code.openBlock(`public static cfnProps2CdkProps(${propertiesArg}: any): any`);
```

Comment:
I think this should be designed as a factory method for the `CfnXxx` resource, not just handle properties.

1. Rename to `fromCloudFormation()`
2. This should accept a full resource entry in CloudFormation, not only `Properties`.
3. Should return the `CfnXxx` type, not `any`.

Reply:
I've done it this way, because a single L1 doesn't have enough context to create itself only from CloudFormation properties, mainly because the logical IDs used in expressions like `Ref`, `Fn::GetAtt`, etc. can change because of `preserveLogicalIds` equal to `false`.

Reply:
Perhaps you can add an optional `resolver` to this which will get a chance to resolve references (something like `JSON.stringify` accepts an optional function to format)

Reply:
Done (I've skipped the part about passing a resolver in favor of not supporting `preserveLogicalIds` = `false` for now, to make this PR simpler - the generated code is quite complex already).